jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletCoreResponse.java

```
@@ -0,0 +1,378 @@
+    }
+
+    @Override
+    public void write(boolean last, ByteBuffer byteBuffer, Callback callback)
+    {
+        if (_included)
+            last = false;
+        try
+        {
+            if (BufferUtil.hasContent(byteBuffer))
+            {
+                if (isWriting())
+                {
+                    String characterEncoding = _response.getCharacterEncoding();
+                    try (ByteBufferInputStream bbis = new ByteBufferInputStream(byteBuffer);
+                         InputStreamReader reader = new InputStreamReader(bbis, characterEncoding))
+                    {
+                        IO.copy(reader, _response.getWriter());
+                    }
+
+                    if (last)
+                        _response.getWriter().close();
+                }
+                else
+                {
+                    BufferUtil.writeTo(byteBuffer, _response.getOutputStream());
+                    if (last)
+                        _response.getOutputStream().close();
+                }
+            }
+
+            callback.succeeded();
+        }
+        catch (Throwable t)
+        {
+            callback.failed(t);
+        }
+    }
+
+    @Override
+    public Request getRequest()
+    {
+        return _coreRequest;
+    }
+
+    @Override
+    public int getStatus()
+    {
+        return _response.getStatus();
+    }
+
+    @Override
+    public void setStatus(int code)
+    {
+        if (_included)
+            return;
+        _response.setStatus(code);
+    }
+
+    @Override
+    public Supplier<HttpFields> getTrailersSupplier()
+    {
+        return null;
+    }
+
+    @Override
+    public void setTrailersSupplier(Supplier<HttpFields> trailers)
+    {
+    }
+
+    @Override
+    public void reset()
+    {
+        _response.reset();
+    }
+
+    @Override
+    public CompletableFuture<Void> writeInterim(int status, HttpFields headers)
+    {
+        return null;
```

I think this should rather throw `UnsupportedOperationException`, even if I don't think this code will ever be executed.